packages/cli/src/commands/setup/auth/templates/dbAuth.auth.ts.template

```diff
@@ -53,24 +53,34 @@ export const hasRole = ({ roles }: { roles: AllowedRoles }): boolean => {
     return false
   }
 
-  if (roles) {
-    if (Array.isArray(roles)) {
-      if (Array.isArray(context.currentUser.roles)) {
-        return context.currentUser.roles?.some((r) => roles.includes(r))
-      } else {
-        roles.some((r) => context.currentUser.roles?.includes(r))
-      }
-    }
+  const currentUserRoles = context.currentUser?.roles
 
-    if (typeof roles === 'string') {
-      return context.currentUser.roles?.includes(roles)
+  if (typeof roles === 'string') {
+    if (typeof currentUserRoles === 'string') {
+      // roleToCheck is a string, currentUser.roles is a string
+      return currentUserRoles === roles
+    } else if (Array.isArray(currentUserRoles)) {
+      // roleToCheck is a string, currentUser.roles is an array
+      return currentUserRoles?.some((allowedRole) => roles === allowedRole)
     }
+  }
 
-    // roles not found
-    return false
+  if (Array.isArray(roles)) {
+    if (Array.isArray(currentUserRoles)) {
+      // roleToCheck is an array, currentUser.roles is an array
+      return currentUserRoles?.some((allowedRole) =>
+        roles.includes(allowedRole)
+      )
+    } else if (typeof context.currentUser.roles === 'string') {
+      // roleToCheck is an array, currentUser.roles is a string
+      return roles.some(
+        (allowedRole) => context.currentUser?.roles === allowedRole
+      )
+    }
   }
 
-  return true
+  // roles not found
+  return false
```

Comment:
Note: has role now returns false when none of the conditions are met. Previously we were returning true - presumably because we were always calling hasRoles from requireAuth, even when roles weren't supplied by the user. 

in Line 105, I now add a check to only run `hasRole` if roles are supplied